data/polynomial.lean

```diff
@@ -750,15 +776,40 @@ else begin
   exact mt leading_coeff_eq_zero.1 h
 end
 
-lemma degree_map_eq [comm_semiring β] [decidable_eq β] (f : α → β) [is_semiring_hom f]
-  (hf : function.injective f) : degree (p.map f) = degree p :=
+lemma subsingleton_of_monic_zero (h : monic (0 : polynomial α)) :
```

Comment:
what about splitting this into `subsingleton (polynomial α) <-> subsingleton α` and `monic 0 -> subsingleton (polynomial α)`